clients/drcachesim/tests/invariant_checker_test.cpp

```diff
@@ -646,14 +661,53 @@ check_duplicate_syscall_with_same_pc()
     return true;
 }
 
+bool
+check_rseq_side_exit_discontinuity()
+{
+    // Negative test: Seemingly missing instructions in a basic block due to rseq side
+    // exit.
+    instr_t *store = XINST_CREATE_store(GLOBAL_DCONTEXT, OPND_CREATE_MEMPTR(REG2, 0),
+                                        opnd_create_reg(REG1));
+    instr_t *move1 =
+        XINST_CREATE_move(GLOBAL_DCONTEXT, opnd_create_reg(REG1), opnd_create_reg(REG2));
+    instr_t *move2 =
+        XINST_CREATE_move(GLOBAL_DCONTEXT, opnd_create_reg(REG1), opnd_create_reg(REG2));
+    instr_t *cond_jmp =
+        XINST_CREATE_jump_cond(GLOBAL_DCONTEXT, DR_PRED_EQ, opnd_create_instr(move2));
+
+    instrlist_t *ilist = instrlist_create(GLOBAL_DCONTEXT);
+    instrlist_append(ilist, cond_jmp);
+    instrlist_append(ilist, store);
+    instrlist_append(ilist, move1);
+    instrlist_append(ilist, move2);
+
+    std::vector<memref_instr_t> memref_instr_vec = {
+        { gen_marker(1, TRACE_MARKER_TYPE_FILETYPE, OFFLINE_FILE_TYPE_ENCODINGS),
+          nullptr },
+        { gen_branch(1), cond_jmp },
+        { gen_instr(1), store },
+        { gen_data(1, false, 42, 4), nullptr },
+        { gen_instr(1), move2 }
```

Comment:
Add a comment calling out the missing move1 so it's clear this is deliberate and in fact the source of the issue being tested; also point out that the jump targets move2 and is the "side exit".